probe_src/libprobe/generated/libc_hooks.c

```diff
@@ -253,7 +407,16 @@ int open(const char *filename, int flags, ...)
     prov_log_try(op);
   }
   size_t varargs_size = ((sizeof(filename)) + (sizeof(flags))) + ((has_mode_arg) ? (sizeof(mode_t)) : (0));
-  int ret = *((int *) __builtin_apply((void (*)()) unwrapped_open, __builtin_apply_args(), varargs_size));
+int ret;
```

Comment:
Does clang on apple not have `__builtin_apply` when you compile from inside the Nix dev shell? Note that the Nix dev shell will smooth over some (not all) of the Linux vs Mac difficulties. Make sure `which gcc` begins with `/nix/store/...`.

Reply:
If not, these changes will have to be propagated "upstream" in `generator/generate_libc_hooks.py` or `generator/libc_hooks_source.c`.